docs/ABI/OldMangling.rst

```diff
@@ -0,0 +1,547 @@
+:orphan:
+
+.. _ABI:
+
+.. highlight:: none
+
+Mangling
+--------
+
+This file documents ONLY the old mangling scheme in use before Swift 4.0.
```

Comment:
I suggest adding something like, "This scheme is still used for the Objective-C class names of Swift classes."

Reply:
Done.